Indicators/Momersion.cs

```diff
@@ -0,0 +1,144 @@
+            : base(name, 3)
+        {
+            _fullPeriod = fullPeriod;
+            _multipliedDiffWindow = new RollingWindow<decimal>(fullPeriod);
+            if (minPeriod < 4)
+            {
+                throw new ArgumentException("The minimum period should be greater of 3.", "minPeriod");
+            }
+            _minPeriod = minPeriod;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MomersionIndicator"/> class.
+        /// </summary>
+        /// <param name="minPeriod">The minimum period.</param>
+        /// <param name="fullPeriod">The full period.</param>
+        public MomersionIndicator(int minPeriod, int fullPeriod)
+            : this("Momersion_" + fullPeriod, minPeriod, fullPeriod)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MomersionIndicator"/> class.
+        /// </summary>
+        /// <param name="fullPeriod">The full period.</param>
+        public MomersionIndicator(int fullPeriod)
+            : this("Momersion_" + fullPeriod, null, fullPeriod)
+        {
+        }
+        
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady
+        {
+            get
+            {
+                if (_minPeriod.HasValue)
+                {
+                    return _multipliedDiffWindow.Count >= _minPeriod;
+                }
+                else
+                {
+                    return _multipliedDiffWindow.IsReady;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resets this indicator to its initial state
+        /// </summary>
+        public override void Reset()
+        {
+            base.Reset();
+            _multipliedDiffWindow.Reset();
+        }
+
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="window"></param>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>
+        /// A new value for this indicator
+        /// </returns>
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
+        {
+            int Mc = 0;
+            int MRc = 0;
+            decimal momersion;
+
+            if (window.Count >= 3) _multipliedDiffWindow.Add((window[0] - window[1]) * (window[1] - window[2]));
+
+            if (this.IsReady)
+            {
+                Mc = _multipliedDiffWindow.Count(obs => obs > 0);
+                MRc = _multipliedDiffWindow.Count(obs => obs < 0);
+                momersion = 100m * Mc / (Mc + MRc);
+            }
+            else
+            {
+                momersion = 50m;
+            }
+
+            Current = new IndicatorDataPoint(input.Time, momersion);
```

Comment:
No need to set `Current` within the `ComputeNextValue` functions on indicators. They're designed such that you just return the new value and the base class will update the current